src/main/java/de/tum/in/www1/artemis/service/hestia/structural/StructuralTestCaseService.java

```diff
@@ -0,0 +1,407 @@
+                    .map(JavaExecutable::getParameters).orElse(Collections.emptyList());
+        }
+        return solutionParameters;
+    }
+
+    private JavaMethod getSolutionMethod(JavaClass solutionClass, StructuralMethod method) {
+        if (solutionClass == null) {
+            return null;
+        }
+        return solutionClass.getMethods().stream().filter(javaMethod -> javaMethod.getName().equals(method.getName())).filter(javaMethod -> {
+            var genericTypes = new ArrayList<>(solutionClass.getTypeParameters());
+            genericTypes.addAll(javaMethod.getTypeParameters());
+            return doParametersMatch(method.getParameters(), javaMethod.getParameters(), genericTypes);
+        }).findFirst().orElse(null);
+    }
+
+    private boolean doParametersMatch(String[] parameters, List<JavaParameter> solutionParameters, List<JavaTypeVariable<JavaGenericDeclaration>> genericDeclarations) {
+        if (parameters == null) {
+            return solutionParameters.isEmpty();
+        }
+        if (parameters.length != solutionParameters.size()) {
+            return false;
+        }
+        for (int i = 0; i < parameters.length; i++) {
+            var typeMatches = parameters[i].equals(solutionParameters.get(i).getType().getValue());
+            var isGeneric = false;
+            for (JavaTypeVariable<JavaGenericDeclaration> type : genericDeclarations) {
+                if (type.getName().equals(solutionParameters.get(i).getType().getValue()) || (type.getName() + "[]").equals(solutionParameters.get(i).getType().getValue())) {
+                    isGeneric = true;
+                    break;
+                }
+            }
+            if (!typeMatches && !isGeneric) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private String generateArgumentsString(String[] parameterTypes, List<JavaParameter> solutionParameters) {
+        String result = "(";
+        if (parameterTypes != null) {
+            for (int i = 0; i < parameterTypes.length; i++) {
+                if (solutionParameters.size() > i) {
+                    // Use original parameter names
+                    parameterTypes[i] = solutionParameters.get(i).getType().getGenericValue() + " " + solutionParameters.get(i).getName();
+                }
+                else {
+                    // Use var[i] as a fallback
+                    parameterTypes[i] = parameterTypes[i] + " var" + i;
+                }
+            }
+            result += String.join(", ", parameterTypes);
+        }
+        result += ")";
+        return result;
+    }
+
+    private StructuralClassElements[] readStructureOracleFile(Path testRepoPath) throws StructuralSolutionEntryGenerationException {
+        try {
+            var testJsonFile = Files.walk(testRepoPath).filter(Files::isRegularFile).filter(path -> "test.json".equals(path.getFileName().toString())).findFirst();
```

Comment:
```suggestion
        try (Stream<Path> files = Files.walk(TestRepoPath)) {
            var testJsonFile = files.filter(Files::isRegularFile).filter(path -> "test.json".equals(path.getFileName().toString())).findFirst();
```
`Files.walk` has to be used inside a try-with-resources to make sure the internal file handles are properly closed.

+2x in the methods below.